mlops/kubeflow/robustness_checker_pytorch/src/robustness_check.py

```diff
@@ -0,0 +1,114 @@
+    parser.add_argument('--nb_classes', type=int,
+                        help='The number of classes of the model', default=2)
+    parser.add_argument('--input_shape', type=str,
+                        help='The shape of one input instance for the pytorch model', default="(1,3,64,64)")
+    parser.add_argument('--feature_testset_path', type=str,
+                        help='Feature test dataset path in the data bucket', default="processed_data/X_test.npy")
+    parser.add_argument('--label_testset_path', type=str,
+                        help='Label test dataset path in the data bucket', default="processed_data/y_test.npy")
+    parser.add_argument('--data_bucket_name', type=str,
+                        help='Bucket that has the processed data', default="training-data")
+    parser.add_argument('--result_bucket_name', type=str,
+                        help='Bucket that has the training results', default="training-result")
+    args = parser.parse_args()
+
+    epsilon = args.epsilon
+    metric_path = args.metric_path
+    model_id = args.model_id
+    robust_status = args.robust_status
+    model_class_file = args.model_class_file
+    model_class_name = args.model_class_name
+    LossFn = args.loss_fn
+    Optimizer = args.optimizer
+    nb_classes = args.nb_classes
+    feature_testset_path = args.feature_testset_path
+    label_testset_path = args.label_testset_path
+    data_bucket_name = args.data_bucket_name
+    result_bucket_name = args.result_bucket_name
+    clip_values = eval(args.clip_values)
+    input_shape = eval(args.input_shape)
+
+    object_storage_url = get_secret('/app/secrets/s3_url', 'minio-service:9000')
+    object_storage_username = get_secret('/app/secrets/s3_access_key_id', 'minio')
+    object_storage_password = get_secret('/app/secrets/s3_secret_access_key', 'minio123')
+
+    metrics = robustness_check(object_storage_url, object_storage_username, object_storage_password,
+                               data_bucket_name, result_bucket_name, model_id,
+                               feature_testset_path=feature_testset_path,
+                               label_testset_path=label_testset_path,
+                               clip_values=clip_values,
+                               nb_classes=nb_classes,
+                               input_shape=input_shape,
+                               model_class_file=model_class_file,
+                               model_class_name=model_class_name,
+                               LossFn=LossFn,
+                               Optimizer=Optimizer,
+                               epsilon=epsilon)
+
+    if not os.path.exists(os.path.dirname(metric_path)):
+        os.makedirs(os.path.dirname(metric_path))
+    with open(metric_path, "w") as report:
+        report.write(json.dumps(metrics))
+
+    robust = "true"
+    if metrics['model accuracy on adversarial samples'] < 0.2:
```

Comment:
The "0.2" should be a value set by the user because there is no single cutoff value for this decision.

Reply:
done